core/client/fs/src/main/java/alluxio/client/file/options/OutStreamOptions.java

```diff
@@ -130,7 +131,11 @@ private OutStreamOptions(ClientContext context, AlluxioConfiguration alluxioConf
     mWriteTier = alluxioConf.getInt(PropertyKey.USER_FILE_WRITE_TIER_DEFAULT);
     mWriteType = alluxioConf.getEnum(PropertyKey.USER_FILE_WRITE_TYPE_DEFAULT, WriteType.class);
     try {
-      mOwner = context.getUserState().getUser().getName();
+      mOwner = AuthenticationUserUtils.getImpersonationUser(
+          context.getUserState().getSubject(), alluxioConf);
```

Comment:
Please add unit tests which verify the impersonation user is got and set and later retrieved by `getOwner()` correctly. You can mock `AuthenticationUserUtils.getImpersonationUser()` in the unit tests.